### DISCUSSION TITLE
changes to keogram.cpp

### DISCUSSION THREAD

**Discussion by EricClaeys** [AndreasLMeg/allsky#4] [Q&A] [answered]
@AndreasLMeg,
A lot of changes have recently been done on Thomas' software, including updates to keogram.cpp that have it use all CPUs and use getopt().

What do your changes to keogram.cpp do, and would you be able to copy them into Thomas' updated version?
Do you have any images that show the difference between your changes enabled and not enabled?

Thanks - Eric Claeys, one of the AllSky developers.

**Comment by AndreasLMeg** [comment, marked as answer]
Hello Eric !
Here I have two pictures that show the differences.

https://www.astro-fotografie.at/allsky2/keograms/keogram-20210614.jpg

https://www.astro-fotografie.at/allsky2/keograms/keogram-20211003.jpg

1. Diagram of the mean values ​​(RedGreenBlue + AVG) Here you can also guess the control algorithm (exposure time and gain)
2. Text with a black background - easier to read
3. The date change is displayed
4. Any column of the original image can be used to create the kegram
5. Additional columns are added to improve the proportions of the image
6. Start parameters are taken from config.sh - endOfNight.sh does not always have to be adjusted

Unfortunately, I currently have no time to quickly and cleanly incorporate these adjustments into the Allsky software. (hobby programmer)

You can see most of the adjustments on my friend's homepage (shutter / gain / keogram)
https://www.astro-fotografie.at/allsky/

Best regards
Andreas
